### PR TITLE
patch(dependencies): update to React 16 and Enzyme 3

### LINKED ISSUE #45: React 16 + Enzyme upgrade
Let's do all these in one shot.

### PR DESCRIPTION
**Pull request by jaebradley**
Resolves #45 

cc: @fysheets 

[`edX-platform Sandbox`](https://jaebradley-react-16.sandbox.edx.org/)

### CODE REVIEW

**Comment by jaebradley** [comment]
I'll fix the code coverage drop

File: src/setupTest.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// TODO: @jaebradley refactor this when Jest issue (https://github.com/facebook/jest/issues/4545#issuecomment-333004504) is patched
```
Comment (arizzitano) [inline src/setupTest.js]: Ooh, this is gnarly but looks like it's the agreed-upon workaround. Will keep an eye on that issue.

File: src/CheckBox/CheckBox.test.jsx [hunk 2]
```diff
@@ -5,12 +5,7 @@ import CheckBox from './index';
 
 describe('<CheckBox />', () => {
   it('attributes are set correctly', () => {
-    const wrapper = mount(
-      <CheckBox
-        name="checkbox"
-        label="check me out!"
-      />,
-    );
+    const wrapper = mount(<CheckBox name="checkbox" label="check me out!" />);
```
Comment (arizzitano) [inline src/CheckBox/CheckBox.test.jsx]: I can definitely see how all-on-one-line is better per airbnb but *sigh* I really liked the old syntax...

File: src/Dropdown/Dropdown.test.jsx [hunk 3]
```diff
@@ -14,20 +14,16 @@ const props = {
 };
 
 const menuOpen = (isOpen, wrapper) => {
-  expect(wrapper.hasClass('show')).toEqual(isOpen);
-  expect(wrapper.find('[type="button"]').prop('aria-expanded')).toEqual(isOpen);
+  expect(wrapper.childAt(0).hasClass('show')).toEqual(isOpen);
```
Comment (arizzitano) [inline src/Dropdown/Dropdown.test.jsx]: Interesting -- so a shallow render no longer returns a direct reference to the DOM node?

File: src/Dropdown/Dropdown.test.jsx [hunk 4]
```diff
@@ -45,21 +41,17 @@ describe('<Dropdown />', () => {
...
-        <Dropdown
-          {...props}
-        />,
-      );
+      wrapper = mount(<Dropdown {...props} />);
     });
 
     it('on toggle click', () => {
-      wrapper.find('[type="button"]').simulate('click');
+      wrapper.find('Button').simulate('click');
```
Comment (arizzitano) [inline src/Dropdown/Dropdown.test.jsx]: Oooh now this is *much* nicer.

File: src/Modal/Modal.test.jsx [hunk 5]
```diff
@@ -134,65 +115,62 @@ describe('<Modal />', () => {
...
     });
 
     it('does nothing on invalid keystroke q', () => {
-      const buttons = wrapper.find('button');
+      let buttons = wrapper.find('button');
 
-      expect(buttons.at(0).matchesElement(document.activeElement)).toEqual(true);
+      expect(buttons.at(0).html()).toEqual(document.activeElement.outerHTML);
       modalOpen(true, wrapper);
+      // buttons.at(0).prop('keyDown')({ key: 'q' });
```
Comment (arizzitano) [inline src/Modal/Modal.test.jsx]: Extra comment? Can we kill this line?

Comment (jaebradley) [reply src/Modal/Modal.test.jsx]: errr this might've been an accidental comment...

alt-text

**Comment by jaebradley** [comment]
## Here's what it looks like when I run my tests locally

image

## The weird thing is...

I actually get to those logical branches, for example [`Modal#61`](https://github.com/edx/paragon/blob/master/src/Modal/index.jsx?utf8=%E2%9C%93#L61)
screen shot 2017-11-20 at 1 08 42 pm

File: src/Modal/Modal.test.jsx [hunk 6]
```diff
@@ -1,12 +1,18 @@
 /* eslint-disable import/no-extraneous-dependencies */
 import React from 'react';
-import { mount } from 'enzyme';
+import Enzyme from 'enzyme';
```
Comment (jaebradley) [inline src/Modal/Modal.test.jsx]: I need to remove this

File: src/Table/Table.test.jsx [hunk 7]
```diff
@@ -1,9 +1,17 @@
 /* eslint-disable import/no-extraneous-dependencies */
 import React from 'react';
-import { shallow, mount } from 'enzyme';
+import Enzyme from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
 
 import Table from './index';
 
+Enzyme.configure({ adapter: new Adapter() });
```
Comment (jaebradley) [inline src/Table/Table.test.jsx]: Remove this as well

File: src/tempPolyfills.test.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
+import raf from './tempPolyfills';
```
Comment (arizzitano) [inline src/tempPolyfills.test.js]: Oh man, you tested the polyfill!! That's dedication right there.

**Pull request merged**